src/sentry/api/bases/avatar.py

```diff
@@ -40,8 +56,12 @@ def get_avatar_filename(self, obj):
         return f"{obj.id}.png"
 
     def put(self, request, **kwargs):
-        obj = kwargs[self.object_type]
-        serializer = AvatarSerializer(data=request.data, context=self.get_serializer_context(obj))
+        obj = kwargs.pop(self.object_type, None)
+        serializer = (
+            SentryAppLogoSerializer(data=request.data, context=self.get_serializer_context(obj))
+            if self.object_type == "sentry_app"
+            else AvatarSerializer(data=request.data, context=self.get_serializer_context(obj))
+        )
```

Comment:
Let's write it like this, I think it might be a tad cleaner

```suggestion
				SerializerCls = AvatarSerializer
				if self.object_type == "sentry_app":
					SerializerCls = SentryAppLogoSerializer
					
        serializer = SerializerCls(data=request.data, context=self.get_serializer_context(obj))
```